src/components/numberInput/index.tsx

```diff
@@ -0,0 +1,172 @@
+   */
+  leadingTextStyle?: StyleProp<ViewStyle>;
+  /**
+   * A trailing text
+   */
+  trailingText?: string;
+  /**
+   * The style of the trailing text
+   */
+  trailingTextStyle?: StyleProp<ViewStyle>;
+  /**
+   * The margin to the left of the input
+   */
+  marginLeft?: number;
+  /**
+   * The margin to the right of the input
+   */
+  marginRight?: number;
+};
+
+function NumberInput(props: NumberInputProps, ref: any) {
+  const {
+    onChangeNumber,
+    initialNumber,
+    fractionDigits = 2,
+    // @ts-expect-error
+    locale = 'en',
+    style,
+    containerStyle,
+    leadingText,
+    leadingTextStyle,
+    marginLeft,
+    trailingText,
+    trailingTextStyle,
+    marginRight,
+    onChangeText,
+    ...others
+  } = props;
+  const [options, setOptions] = useState<Options>(generateOptions(locale, fractionDigits));
+  const [data, setData] = useState<NumberInputResult>();
+
+  useDidUpdate(() => {
+    setOptions(generateOptions(locale, fractionDigits));
+  }, [locale, fractionDigits]);
+
+  const handleInitialValueChange = () => {
+    const newData = getInitialResult(options, initialNumber);
+    onChangeNumber(newData);
+    setData(newData);
+  };
+
+  useEffect(() => {
+    handleInitialValueChange();
+  }, []);
+
+  useDidUpdate(() => {
+    handleInitialValueChange();
+  }, [initialNumber]);
+
+  const hasText = useMemo(() => {
+    // Render both leading and trailing accessories so the text is centered AND the margin between the text and the accessories is correct
+    return !isEmpty(leadingText) || !isEmpty(trailingText);
+  }, [leadingText, trailingText]);
+
+  const leadingAccessoryStyle: TextProps['style'] = useMemo(() => {
+    return [{textAlign: 'right'}, leadingTextStyle, marginLeft ? {marginLeft} : undefined];
+  }, [leadingTextStyle, marginLeft]);
+
+  const leadingAccessory = useMemo(() => {
+    if (hasText) {
+      return <Text style={[styles.accessory, leadingAccessoryStyle]}>{leadingText}</Text>;
+    }
+  }, [hasText, leadingText, leadingAccessoryStyle]);
+
+  const trailingAccessoryStyle: TextProps['style'] = useMemo(() => {
+    return [trailingTextStyle, marginRight ? {marginRight} : undefined];
+  }, [trailingTextStyle, marginRight]);
+
+  const trailingAccessory = useMemo(() => {
+    if (hasText) {
+      return <Text style={[styles.accessory, trailingAccessoryStyle]}>{trailingText}</Text>;
+    }
+  }, [hasText, trailingText, trailingAccessoryStyle]);
+
+  const _style = useMemo(() => {
+    return [style, leadingAccessory ? undefined : {marginLeft}, trailingAccessory ? undefined : {marginRight}];
+  }, [style, leadingAccessory, marginLeft, trailingAccessory, marginRight]);
+
+  const _containerStyle = useMemo(() => {
+    return [styles.containerStyle, containerStyle];
+  }, [containerStyle]);
+
+  const _onChangeText = useCallback((text: string) => {
+    const newData = parseInput(text, options);
+    onChangeNumber(newData);
+    setData(newData);
+    onChangeText?.(text);
```

Comment:
Do we need both `onChangeNumber` and `onChangeText`? 
maybe we can omit onChangeText from the type..